tests/curators/test_curators_examples.py

```diff
@@ -563,33 +563,62 @@ def test_anndata_curator_varT_curation_legacy(ccaplog):
             varT_schema.delete(permanent=True)
 
 
-def test_soma_curator(curator_params: dict[str, str | FieldAttr]):
-    """Test SOMA curator implementation."""
+def test_anndata_curator_nested_uns():
+    """Test AnnDataCurator with nested uns slot validation."""
     adata = datasets.small_dataset1(otype="AnnData")
-    tiledbsoma.io.from_anndata(
-        "./small_dataset1.tiledbsoma", adata, measurement_name="RNA"
-    )
+    adata.uns["study_metadata"] = {"temperature": 21.6, "experiment_id": "EXP001"}
 
-    curator = ln.Curator.from_tiledbsoma(  # type: ignore
-        "./small_dataset1.tiledbsoma",
-        var_index={"RNA": ("var_id", bt.Gene.ensembl_gene_id)},
-        **curator_params,
-    )
-    artifact = curator.save_artifact(key="examples/dataset1.tiledbsoma")
+    uns_schema = ln.Schema(
+        features=[
+            ln.Feature(name="temperature", dtype=float).save(),
+            ln.Feature(name="experiment_id", dtype=str).save(),
+        ],
+    ).save()
 
-    assert set(artifact.features.get_values()["cell_type_by_expert"]) == {
-        "CD8-positive, alpha-beta T cell",
-        "B cell",
-    }
-    assert set(artifact.features.get_values()["cell_type_by_model"]) == {
-        "T cell",
-        "B cell",
-    }
+    anndata_schema = ln.Schema(
+        otype="AnnData",
+        slots={
+            "uns:study_metadata": uns_schema,
+        },
+    ).save()
 
-    assert artifact._key_is_virtual
+    curator = ln.curators.AnnDataCurator(adata, anndata_schema)
+    assert isinstance(curator.slots["uns:study_metadata"], ln.curators.DataFrameCurator)
+
+    curator.validate()
+    artifact = curator.save_artifact(key="examples/anndata_with_uns.h5ad")
+
+    assert artifact.schema == anndata_schema
+    assert "uns:study_metadata" in artifact.features.slots
+    assert artifact.features.slots[
+        "uns:study_metadata"
+    ].features.first() == ln.Feature.get(name="temperature")
+
+    with pytest.raises(InvalidArgument) as e:
+        adata = datasets.small_dataset1(otype="AnnData")
+        bad_schema = ln.Schema(
+            otype="AnnData",
+            slots={"uns:nonexistent": uns_schema},
+        ).save()
+        ln.curators.AnnDataCurator(adata, bad_schema)
+    assert (
+        "Schema slot 'uns:study_metadata' specifies path uns['study_metadata'] but key 'study_metadata' not found."
+        in str(e.value)
+    )
+
+    with pytest.raises(InvalidArgument) as e:
+        bad_schema = ln.Schema(
+            otype="AnnData",
+            slots={"uns:temperature:nonexistent_nested": uns_schema},
+        ).save()
+        ln.curators.AnnDataCurator(adata, bad_schema)
+    assert "key 'study_metadata' not found" in str(e.value)
```

Comment:
We have to pay attention to the word choice here and double-check that this is consistent with what pandera and lamindb issue otherwise.

"key `study_metadata` not found" sounds to me like there is a `study_metadata` key in the dictionary but it's not found in the schema (I think that intuition comes from FileNotFound error, KeyError)

I think pandera says "column xyz is missing in dataframe" -- I think a similar word choice would be clearer